packages/iris-grid/src/IrisGrid.jsx

```diff
@@ -694,6 +694,10 @@ export class IrisGrid extends Component {
     getFormatColumns(columns, rules)
   );
 
+  // customColumns arg is needed to invalidate the cache
+  // eslint-disable-next-line no-unused-vars
+  getCachedModelColumns = memoize((model, customColumns) => model.columns);
```

Comment:
I don't think I understand why this is needed? The `get columns` in the model is already memoized

Reply:
There was a render loop that I tracked down to `get columns` cache being invalidated on each `applyCustomColumns` call except when the arg is an empty array. This in turn invalidates `getCachedPreviewFormatColumns` and `getCachedFormatColumns` cache, triggers `applyCustomColumns` again and so on...
Confirmed on the irisapi page:
```
currentTable.applyCustomColumns(["x=1"])
cols = currentTable.columns
currentTable.applyCustomColumns(["x=1"])
cols === currentTable.columns
>>> false
```

Reply:
Ah sounds good